trpc/router/goal.ts

```diff
@@ -1617,4 +1696,70 @@ export const goal = router({
                 });
             }
         }),
+    getConnectedGoalsByCriteria: protectedProcedure.input(z.object({ id: z.string() })).query(async ({ input }) => {
+        try {
+            const criteriaListByGoal = await prisma.goalAchieveCriteria.findMany({
+                where: {
+                    goalIdAsCriteria: input.id,
+                    AND: {
+                        OR: [{ deleted: null }, { deleted: false }],
+                    },
+                },
+                include: {
+                    goal: {
+                        include: {
+                            state: true,
+                        },
+                    },
+                },
+            });
+
+            return criteriaListByGoal;
+        } catch (error: any) {
+            throw new TRPCError({ code: 'INTERNAL_SERVER_ERROR', message: error.message, cause: error });
+        }
+    }),
+    getGoalCriteriaList: protectedProcedure.input(z.object({ id: z.string().optional() })).query(async ({ input }) => {
+        if (!input.id) {
+            return;
+        }
+
+        try {
+            const criteriaList = await prisma.goalAchieveCriteria.findMany({
+                where: {
+                    AND: [
+                        { goalId: input.id },
+                        {
+                            OR: [{ deleted: false }, { deleted: null }],
+                        },
+                    ],
+                },
+            });
+
+            return criteriaList;
+        } catch (error: any) {
+            throw new TRPCError({ code: 'INTERNAL_SERVER_ERROR', message: error.message, cause: error });
+        }
+    }),
+    checkGoalInExistingCriteria: protectedProcedure
+        .input(
+            z
+                .object({
+                    currentGoalId: z.string(),
+                    selectedGoalId: z.string(),
+                })
+                .refine(async (val) => {
+                    const criteria = await prisma.goalAchieveCriteria.findFirst({
+                        where: {
+                            AND: {
+                                goalId: val.selectedGoalId,
+                                criteriaGoalId: val.currentGoalId,
+                            },
+                        },
+                    });
+
+                    return criteria == null;
+                }, 'These bindings is already exist'),
+        )
+        .mutation(async () => {}),
```

Comment:
It looks more like a query, not a mutation. Queries can also be called imperatively
```
const utils = trpc.useContext();
await utils.goal.checkGoalInExistingCriteria.fetch()
```